src/main/java/com/openlattice/edm/PostgresEdmManager.java

```diff
@@ -186,20 +198,40 @@ public void revoke(
         }
     }
 
-    public Set<UUID> getAllPropertyTypeIds() {
-        String getAllPropertyTypeIds = String.format("SELECT id from %1$s", PROPERTY_TYPES);
+    /**
+     * Duplicate of
+     * {@link com.openlattice.data.storage.PostgresEntityDataQueryService#getLinkingEntitySetIdsOfEntitySet(UUID)}
+     */
+    public Set<EntitySet> getAllLinkingEntitySetsForEntitySet( UUID entitySetId ) {
+        return entitySets.values().stream()
```

Comment:
This pulls all entity sets locally and then filters. Look at ```PermissionMapstore``` and ```HazelcastAuthorizationService``` for an example of how to use Hazelcast index to this quickly and efficiently. You'll want to override ```getMapConfig()``` so that it adds a map index config ```linkedEntitySets[any]``` that is not ordered. This will allow you to use ```entitySets.value(...)``` to do an indexed predicate query that will only retrieve the entity sets satisfying the query with the filtering done at all the nodes.